closure-typing/calculator.js

```diff
@@ -1,5 +1,21 @@
-function calculator() {
-  // your code here
+function calculator(str) {
+  var mathFuncList = {
+    '+' : (a, b) => {return +a + +b},
+    '-' : (a, b) => {return a - b},
+    '*' : (a, b) => {return a * b},
+    '/' : (a, b) => {return a / b}
+  }
+  arr = str.replace(/([+\-*/])/g, ' $1 ').split(' ')
+
+  while (arr.length > 1){
+    i = arr.indexOf('*')
+    i < 0 ? i = arr.indexOf('/') : i
+    i < 0 ? i = arr.indexOf('+') : i
+    i < 0 ? i = arr.indexOf('-') : i
```

Comment:
Oh, I figure it out, what is strange about it, it should be looking smth like this
 `i = i < 0 ? arr.indexOf('-') : i`

Now you getting right the way that you changing `i` on some condition